src/App.tsx

```diff
@@ -17,18 +46,44 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                todos={todos}
+                changeFilters={getVisibleGoods}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {visibleTodos.length > 0 ? (
+                <TodoList
+                  todos={visibleTodos}
+                  selectedTodo={selectedTodo}
+                  onTodoSelect={handleUserSelect}
+                />
+              ) : (
+                <>
+                  {isDownloadComplete ? (
+                    <>
+                      <h1 className="is-vcentered">
+                        No todos fulfilling your request
+                      </h1>
+                    </>
+                  ) : (
+                    <Loader />
+                  )}
+                </>
```

Comment:
Why do you need this fragment?

```suggestion
              {isDownloadComplete ? (
                <>
                  <h1 className="is-vcentered">
                    No todos fulfilling your request
                  </h1>
                </>
              ) : (
                <Loader />
              )}
```

Reply:
for any reason my linter uses bad words when I am trying to use JS `{ }` right in `( )` parenthesis. I tried several alternatives but this one was the clearest and shortest among all of them

Reply:
so to ue something like if inside else statement I have to buuild this construction:
`{true ? ( ) : (
  <>
    { true ? ( ) : ( ) }
  </>
}`

Reply:
But I think these are extra after all 
```suggestion
                <>
                  {isDownloadComplete ? (
                      <h1 className="is-vcentered">
                        No todos fulfilling your request
                      </h1>
                  ) : (
                    <Loader />
                  )}
                </>
```